include/irr/asset/IDescriptorSetLayout.h

```diff
@@ -0,0 +1,45 @@
+#ifndef __IRR_I_DESCRIPTOR_SET_LAYOUT_H_INCLUDED__
+#define __IRR_I_DESCRIPTOR_SET_LAYOUT_H_INCLUDED__
+
+#include "irr/asset/ShaderCommons.h"
+#include "irr/asset/SSamplerParams.h"
+
+namespace irr { namespace asset
+{
+
+enum E_DESCRIPTOR_TYPE
+{
+    EDT_SAMPLER = 0,
+    EDT_COMBINED_IMAGE_SAMPLER = 1,
```

Comment:
we can support this in two ways:
- Descriptor set with immutable sampler
- Descriptor set with combined image+sampler

Reply:
@Crisspl only this enum can stay, the one above and below must go until transition to Vulkan-only

Reply:
resolved